### PR TITLE
Restructure & add pre-built examples

### PR DESCRIPTION
**Pull request by ellahathaway**
Related to https://github.com/dotnet/source-build/issues/3358#issuecomment-1542661513

This PR aims to improve the readability and usability of the pre-built elimination document. It accomplishes the following:
- Improve the document's readability and usability by providing clear, step-by-step instructions on how to handle new dependencies and pre-builts in the source-build process.
- The document now explicitly lists the possible actions that a developer might need to take, such as updating versions of certain packages, modifying Version.Details.xml and Versions.Props files, and adding new packages to the source-build-reference-packages or source-build-externals repositories.
- Each action is explained in detail, with specific examples given. This makes it easier for developers to understand what they need to do and how to do it.
- The document also provides links to additional resources, such as the source-build-reference-packages and source-build-externals repositories, which can provide further information and context.
- The document mentions the importance of comparing against SourceBuildPrebuiltBaseline.xml file.
- Splits important information into separate paragraphs and sections to effectively point readers to key information and allow readers to quickly identify what they need to do.

These formatting changes, along with the content updates, make the document more user-friendly and accessible to developers of different experience levels.

### CODE REVIEW

**Comment by mthalman** [comment]
@ViktorHofer - I've added you as a review as you encounter prebuilts fairly often. It'd be good to get your perspective since the intention is that you'd be able to follow these instructions to resolve such issues on your own.

**Comment by ViktorHofer** [comment]
Don't have time to review right now but will take a look over the weekend or on Monday.

File: Documentation/eliminating-pre-builts.md
Comment (mthalman) [inline Documentation/eliminating-pre-builts.md:181]: Somewhere we need to state that adding a package to SBRP isn't always the right step to take to resolve a prebuilt. For example, the repo may have added a dependency on a package and included that in Version.Details and it will give them a detected prebuilt error. Then they may think it needs to be added to SBRP. But in the context of the VMR, if the dependency comes from another repo that gets built before their repo, the package in SBRP is not needed because the VMR will flow the live version. In this case we don't want it added to SBRP and it should rather be marked as an exclusion in the baseline. I feel this is an important scenario that needs to be described.

Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md:181]: I'm wondering if this should be added as part of https://github.com/dotnet/source-build/blob/main/Documentation/sourcebuild-in-repos/new-dependencies.md#adding-dependencies  and then pointed to in this document as a short warning/note about the importance of building the repo in the VMR when detecting prebuilts. The reasoning behind this suggestion is that the eliminating document suggests steps to determine the causes of prebuilts, but it does not go into depth on how to resolve them. Instead, I was leaving that work up the adding-dependencies documentation.

As you pointed out in a different comment above, that documentation does need refactoring. In applying this suggestion, I think that the updates to the adding-dependencies doc can be part of this PR.

File: Documentation/eliminating-pre-builts.md
Comment (ViktorHofer) [inline Documentation/eliminating-pre-builts.md:78]:
```suggestion
    1. If you have detected a new pre-built that is not in the current baseline, `./eng/SourceBuildPrebuiltBaseline.xml`, continue to step 3.
```

File: Documentation/eliminating-pre-builts.md
Comment (ViktorHofer) [inline Documentation/eliminating-pre-builts.md:308]: Many repositories in the source-build graph already use NuGet's Central Package Management feature. It would be good to mention that CPM can help with lifting and unifying transitive dependencies via the transitive pinning feature: https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management#transitive-pinning

**Review by ViktorHofer — APPROVED** [review]

**Review by MichaelSimons — COMMENTED** [review]
Ella,  Thanks for improving this documentation!

File: Documentation/eliminating-pre-builts.md
Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md:53]: `Dangling dependencies` was part of the old documentation, so I'm unsure if this is something the infra is actually able to classify.

File: Documentation/eliminating-pre-builts.md
Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md:55]: That's a good point. I'll change the title of the section.

File: Documentation/eliminating-pre-builts.md
Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md:92]: Good point. I'll remove this section.

File: Documentation/eliminating-pre-builts.md
Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md:139]: The baseline is described in the `Allowed Exceptions` section. Should I move that section up in the document?

File: Documentation/eliminating-pre-builts.md
Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md:175]: I think that these steps would be better explained in depth in the adding dependencies documentation but including them as options in the example is helpful. It would be up to the reader to read the adding dependencies documentation to determine which step to take.

File: Documentation/eliminating-pre-builts.md
Comment (MichaelSimons) [reply Documentation/eliminating-pre-builts.md]: I can't find anything in the code that maps to this.  Can you check to see if I missed anything and then remove this doc if you don't find anything?

File: Documentation/eliminating-pre-builts.md
Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md:203]: Actually, based on Matt's comment above, I think that these steps would be better served in the dependencies documentation, as you originally suggested. My thinking is also that if the steps ever changed, we'd have to alter documentation in two places, rather than one. I'm going to remove the examples.

File: Documentation/eliminating-pre-builts.md
Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md]: Added here: https://github.com/dotnet/source-build/pull/4043/commits/010a2cca0ff2703d2884e1ed017123b41f216e57#diff-d4752b9dd717bdd56774b2d6f54ec8365f9cafd10917354a454278a917f0a104R93

File: Documentation/eliminating-pre-builts.md
Comment (ellahathaway) [reply Documentation/eliminating-pre-builts.md:154]: Updated the document in https://github.com/dotnet/source-build/pull/4043/commits/010a2cca0ff2703d2884e1ed017123b41f216e57. I left SBRP at the bottom it felt orthogonal to the rest of the cases in the document (i.e. the prior cases tend to flow into each other). Let me know if you think I should still move it up.

**Comment by ellahathaway** [comment]
@ViktorHofer - Requesting review since I've now updated 'new-dependencies.md' As Matt mentioned in a previous comment, your perspective is important since we intend for readers to be able to follow the instructions and resolve issues on their own. Thanks in advance!

**Comment by ViktorHofer** [comment]
Sorry for the late reply. I still haven't read this but please don't wait on my approval with merging this. I will file an issue if I find anything that needs an update. Thanks for working on this.

**Comment by ellahathaway** [comment]
Closing this in favor of other higher-priority work. Keeping the branch open for when there's more time to return to this or if someone else would like to pick up the changes.

**Pull request closed without merging**
